Kernel/Arch/ProcessorFunctions.h

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
To distinguish this "header file that should be included as implementation details of a template class" from a normal header file, we should use one of the common extensions for this purpose. take your pick from `.inc`, `.inl`, etc.